src/stores/__tests__/locale.test.ts

```
@@ -154,10 +161,26 @@ describe('the reducer', () => {
         expect(theNewStore.loading).toBe(true);
     });
 
-    it('should return store unchanged if action is undefined', () => {
-        const theOriginalStore = buildStoreWithLocale(aLocale);
-        const theNewStore = locale.reducer(theOriginalStore, undefined);
-        expect(theNewStore.code).toBe(theOriginalStore.code);
+    it('when called with LOAD_CURRENTLOCALE_SUCCESS should return store with locale code from action', () => {
+        const theStore = buildStoreLoadingLocale();
+        const theAction = {
+            type: constants.LOAD_CURRENT_LOCALE_SUCCESS as typeof constants.LOAD_CURRENT_LOCALE_SUCCESS,
+            payload: { locale: aLocale },
+        };
+        const theNewStore = locale.reducer(theStore, theAction);
+        expect(theNewStore.code).toBe(theAction.payload.locale.code);
+        expect(theNewStore.loading).toBe(false);
+    });
+
+    it('when called with LOAD_CURERNT_LOCALE_FAILURE should return store with loading flag set false', () => {
+        const errorMessage = aString();
+        const theStore = buildStoreLoadingLocale();
+        const theAction = {
+            type: constants.LOAD_CURRENT_LOCALE_FAILURE as typeof constants.LOAD_CURRENT_LOCALE_FAILURE,
+            payload: { message: errorMessage, locale: aLocale },
+        };
+        const theNewStore = locale.reducer(theStore, theAction);
+        expect(theNewStore.loading).toBe(false);
```

What happens with the error message, is it part of the store after?

---

It is not currently being put in the store (see locale reducer). I avoided tackling this until we have a clear path for handling this sort of thing. Maybe this should be outlined/discussed in a followup issue?

As noted below, unless we can think of something better, the least we can do is display a message to the user indicating that something went wrong.

---

We should think about error handling. Until then, TDD should encourage us to only write code that has observable (i.e. testable) effects. So, if we were using TDD, which I encourage, then the error message would not be there in the test or production code. This helps us focus our attention on always writing code that has value to our users. I'm not asking for any changes now, just using this as an opportunity to plug TDD. I don't tend to pass up on those much ;-)

---

That's a fair point. At the time of writing, my thinking was that not handling exceptions (or worse; silently suppressing them) is best avoided because they can easily be forgotten (hence the `console.error()` popping up in tests right now). In the future I'm wondering whether it would be better to just not include the `try { ... } catch(e) { ... }` so that any exceptions cause application to crash? (that way we're less likely to ignore/forget about it and we can add the error handling when we have a plan for how to implement it).

In saying all that, the error handling in this piece of functionality (`loadCurrentLocale()` and `setLocale()`) is incomplete and could definitely be improved.